recipes/openjdk/all/test_package/conanfile.py

```diff
@@ -0,0 +1,22 @@
+from conans import ConanFile
+from conans.errors import ConanException
+from io import StringIO
+
+required_conan_version = ">=1.36.0"
```

Comment:
What happens if you're running an older version? The main recipe works and then it errors out during test?

Reply:
It raises a `ConanException` error: https://docs.conan.io/en/latest/reference/conanfile/other.html#requiring-a-conan-version-for-the-recipe

Reply:
Yes I know it raises an error, but I'm talking about the placement.
I usually see this line in the main conanfile.py. By having it on the test_package's conanfile.py, would you allow the package to build and then error out during test?
Are both conanfiles evaluated before build?

Reply:
Nice catch

Reply:
test_package is not evaluated before the build

Reply:
so should I add the `required_conan_version` to the main conanfile ?

Reply:
I think you are correct now.
The main recipe needs conan 1.33.
The test package needs conan 1.36.

Reply:
it does not make sense to require two different versions. lets use 1.36 in main recipe

Reply:
You can build the main recipe without a test recipe (`-tf None`) or build with a different test recipe (`-tf other_test_package`).

Reply:
This is ok. For a client using `1.33` it will fail in a `conan create` command, but it is accurate and, as said before, you can avoid running these tests, or use a different test folder... and a `conan install ... --build=openjdk` won't fail, which is the command you would run when retrieving this recipe from a remote (`test_package` is not uploaded to the remote).